apps/sms/js/thread_list_ui.js

```diff
@@ -394,19 +395,24 @@ var ThreadListUI = {
   },
 
   markReadUnread: function thlui_markReadUnread(selected, isRead) {
-    selected.forEach((id) => {
+    var filterIds = (id) => {
       var thread  = Threads.get(id);
       var markable = thread && (!thread.hasDrafts || isRead);
 
       if (markable) {
         thread.unreadCount = isRead ? 0 : 1;
         this.mark(thread.id, isRead ? 'read' : 'unread');
-
-        MessageManager.markThreadRead(thread.id, isRead);
       }
-    });
+      return markable;
+    };
 
     this.cancelEdit();
+    this.performMarkReadUnread(selected.filter(filterIds), isRead);
+  },
+
+  performMarkReadUnread:
+    function thlui_performMarkReadUnread(threadToMark, isRead) {
```

Comment:
nit: put the "function" part on the same column than "performMarkReadUnread", I think this is a little more readable.